modules/ROOT/pages/integrations/expose-metrics-to-prometheus.adoc

```diff
@@ -1 +1,179 @@
++
+--
+.Dependency to be added to the `pom.xml` file to enable metrics
+[source,xml]
+----
+<dependency>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-addons-quarkus-monitoring-prometheus</artifactId>
+</dependency>
+----
+--
+
+. Rebuild your workflow application.
++
+The metrics is available at `/q/metrics` endpoint.
+
+[[con-consume-metrics-sw]]
+== Metrics consumption in {context}
+
+After enabling the metrics in {context}, the generated metrics can be consumed from OpenShift, Kubernetes, and Prometheus to visualize on different dashboard tools.
+
+[[proc-consume-metrics-openshift]]
+=== Consuming metrics from OpenShift
+
+If your workflow server is running on OpenShift, then you can use the server to monitor your workflow application. Also, you can perform the task of consuming metrics from OpenShift.
+
+.Prerequisites
+* Metrics is enabled in {context}.
++
+For more information, see <<proc-enable-metrics-sw, Enabling metrics in {context}>>.
+
+.Procedure
+. To consume metrics from OpenShift, enable monitoring for user-defined projects. 
++
+--
+For more information, see link:{openshift_monitoring_url}[Enabling monitoring for user-defined projects] in OpenShift documentation.
+
+When you enable monitoring for user-defined projects, the Prometheus Operator is installed automatically. 
+--
+
+. Create a service monitor as shown in the following configuration:
++
+--
+.Example configuration in `service-monitor.yaml`
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: prometheus-app-monitor
+  name: prometheus-app-monitor
+  namespace: my-project
+spec:
+  endpoints:
+  - interval: 30s
+    targetPort: 8080
+    path: /q/metrics
+    scheme: http
+  selector:
+    matchLabels:
+      app-with-metrics: 'serverless-workflow-app'
+----
+--
+
+. Run the following command to apply the service monitor:
++
+--
+.Apply service monitor
+[source,shell]
+----
+oc apply -f service-monitor.yaml
+----
+--
+
+In the previous procedure, a service monitor named `prometheus-app-monitor` is created, which selects applications containing the label as `app-with-metrics: serverless-workflow-app`. Ensure that your workflow application contains the same label.
+
+After that, Prometheus sends request to the `/q/metrics` endpoint for all the services that are labeled with `app-with-metrics: serverless-workflow-app` every 30 seconds. For more information about monitoring Quarkus application using Micrometer and Prometheus into OpenShift, see link:{openshift_micrometer_url}[Quarkus - Micrometer Metrics].
+
+[IMPORTANT]
+====
+Consuming metrics from Kubernetes is similar to OpenShift. However, you need to install the Prometheus Operator project manually. 
+
+For more information about installing Prometheus Operator, see link:{prometheus_operator_url}[Prometheus Operator] website.
+====
```

Comment:
Not sure that this should be in. We should leverage Openshift.

Reply:
Will do.